docs/kilnjs.md

```diff
@@ -0,0 +1,220 @@
+
+```
+schema['_groups'].settings['_placeholder'].height = '200px';
+```
+
+However, note that the JSON that is returned still needs to conform to the structure needed by Kiln.  Deleting properties or adding properties that Kiln doesn't recognize can and will result in errors.
+
+
+---
+
+## KilnInput
+
+The real power of Kilnjs comes from the KilnInput object, which can be used to make the fields within a schema truly dynamic.  Using KilnInput you can add events to the different form inputs as detailed on the [Form Inputs page](input).  You can also subscribe to Vuex actions as described below.
+
+To make a schema field a KilnInput, you set it to a new instance of KilnInput, passing it to the schema and the name of the field.
+
+```
+const KilnInput = window.kiln.kilnInput;
+
+module.exports = (schema) => {
+  schema.title = new KilnInput(schema, 'title');
+
+  return schema;
+};
+```
+
+---
+
+## Events
+
+You can attach events to the schema inputs.  The events you can attach vary depending upon the type input. The different events are detailed on the [Form Inputs page](input).
+
+* ***on(event, callbackFunction)*** - pass it the event as a string, along with a callback function that is run after the event happens
+
+```
+  schema.enableSocialButtons.on('input', (val) => {
+    if(val) {
+      schema.shareServices.show();
+    } else {
+      schema.shareServices.hide();
+    }
+  });
+```
+
+### Validation
+
+By attaching events, one of the things you can use kilnjs for is to do field validation.
+
+```
+module.exports = (schema) => {
+  schema.title = new KilnInput(schema, 'title');
+
+  schema.title.on('keydown', (e) => {
+    // prevent numbers from being entered in the title field
+    if (!isNaN(e.key)) {
+      e.preventDefault();
+    }
+  });
+
+  return schema;
+};
+```
+
+---
+
+## Vuex Actions to Subscribe To
+
+The following are some of the Vuex actions that you can subscribe to using Kiln.js. The actions can be scoped so that they only trigger the subscription function when the action is triggered by a component of the same type as the subscribing component. In other words, if scope is set to true on a paragraph element, then the subscription function would only be called when a paragraph element triggers the Vuex action, but not when any other type of component triggers the Vuex action.  When the scope is set to false, the subscription function is called whenever _any_ component triggers the Vuex action.
+
+* ***OPEN_FORM*** - Triggered when a Kiln form is opened. Note that this can triggered by both a modal form _and_ an inline form, so just clicking into a paragraph or other inline field will trigger this action. The payload returned from this action contains a JSON representation of the Schema along with the component uri that was opened in the form.
```

Comment:
```suggestion
* ***OPEN_FORM*** - Triggered when a Kiln form is opened. Note that this can be triggered by both a modal form _and_ an inline form, so just clicking into a paragraph or other inline field will trigger this action. The payload returned from this action contains a JSON representation of the Schema along with the component URI that was opened in the form.
```

Reply:
👍